pylint/reporters/ureports/nodes.py

```diff
@@ -14,30 +14,33 @@
 
 A micro report is a tree of layout and content objects.
 """
+from typing import Any, List, Optional, Tuple, Union
+
+from pylint.reporters.ureports.text_writer import TextWriter
 
 
 class VNode:
-    def __init__(self, nid=None):
+    def __init__(self, nid: Optional[Any] = None) -> None:
         self.id = nid
         # navigation
-        self.parent = None
-        self.children = []
-        self.visitor_name = self.__class__.__name__.lower()
+        self.parent: Optional[Any] = None
```

Comment:
This should be `Optional[VNode]`. But this will create a `mypy` warning as `VNode` does not have a `parent` attribute. See L79.
Using `Optional[BaseLayout]` would create problems with `append()` and `insert()`. However, I think those can be moved to the `BaseLayout` class as `Text` and `VerbatimText` should never use those methods. This would allow typing with `Optional[BaseLayout]`.

Reply:
```suggestion
        self.parent: Optional["BaseLayout"] = None
```
After #5015 this should now be possible